src/backends/beamcoder.ts

```diff
@@ -142,717 +92,365 @@ const getFilter = async({
-                        this.filteredFrames = [frame, ...this.filteredFrames];
-                        return false;
-                    }
-                    else {
-                        LOG_SINGLE_FRAME_DUMP_FLOW && console.log('frame.pts > targetPts | ', `${frame.pts} > ${targetPts} - but no last frame available - resolving with frame!`);
-                        // Add this frame back to the buffer - we don't need it yet.
-                        this.filteredFrames = [frame, ...this.filteredFrames];
-                        this.lastFrame = frame;
-                        resolve(frame);
-                        return false;
-                    }
-                }
-
-                if (this.packet === null && this.filteredFrames.length === 0 && targetPts >= this.lastFrame.pts) {
-                    LOG_SINGLE_FRAME_DUMP_FLOW && console.log(`Last frame has been reached, resolving with last frame, pts=${this.lastFrame.pts}`);
-                    resolve(frame);
-                    return false;
-                }
-
-                LOG_SINGLE_FRAME_DUMP_FLOW && console.log('frame.pts < targetPts | ', `${frame.pts} < ${targetPts} - continuing`);
-
-                this.lastFrame = frame;
-
-                return true;
-            };
-
-            finishReadingCleanlyPromise = this.readFrames({
-                onFrameAvailable,
-                flush: true,
-            });
-
-            await finishReadingCleanlyPromise;
-        });
-
-        await finishReadingCleanlyPromise;
-
-        return frame;
-    }
-
-
-    /**
-     * Seek to a given time in the stream
-     */
-    async seekToTime(targetTime: number) {
-        await this.seekToPTS(this.timeToPts(targetTime));
-    }
-
-    /**
-     * Convert a time (in seconds) to PTS (based on timebase)
-     */
-    timeToPts(time: number) {
-        const time_base = this.demuxer.streams[this.#streamIndex].time_base;
+    _timeToPTS(time: number) {
+        const time_base = this.#demuxer.streams[this.#streamIndex].time_base;
         return time * time_base[1] / time_base[0];
     }
 
     /**
-     * Convert a PTS (based on timebase) to PTS (in seconds)
+     * Get the time in seconds from a given presentation timestamp (PTS)
      */
     ptsToTime(pts: number) {
-        const time_base = this.demuxer.streams[this.#streamIndex].time_base;
+        const time_base = this.#demuxer.streams[this.#streamIndex].time_base;
         return pts * time_base[0] / time_base[1];
     }
 
     /**
-     * This just calls `this.demuxer.read()` with additional logging in case of errors.
+     * Get the frame at the given presentation timestamp (PTS)
      */
-    async #readPacketWrapped() {
-        if (this.#streamStopped) {
-            throw new Error('Error: Trying to read after stream has stopped.');
+    async _getFrameAtPts(targetPTS: number) {
+        VERBOSE && console.log('_getFrameAtPts', targetPTS, '-> duration', this.duration);
```

Comment:
This function is pretty dense, I wish we break it down later. not something to worry about now though

Reply:
I agree, once we're feature complete, I'll do another pass